app/concepts/tickets/use_cases/create_for_reservation.rb

```diff
@@ -0,0 +1,43 @@
+module Tickets
+  module UseCases
+    class CreateForReservation
+      SeatsNotAvailableError = Class.new(StandardError)
+
+      attr_reader :reservation, :screening, :tickets_params
+
+      def initialize(reservation:, screening:, tickets_params: [])
+        @tickets_params = tickets_params
+        @reservation = reservation
+        @screening = screening
+      end
+
+      def call
+        Ticket.transaction do
+          raise SeatsNotAvailableError, "Every seats are taken" if available_seats.empty?
+          raise SeatsNotAvailableError, "Provided seat are not available" unless available?
+
+          tickets_params.each do |params|
+            ticket = reservation.tickets.create(params)
+            raise SeatsNotAvailableError, "Provided parameters are invalid" if ticket.invalid?
```

Comment:
The same thing as here: https://github.com/kodowicz/monte-cinema/pull/5/files#r646533780

If record was not saved due to other problem that it being invalid this will not be risen.